src/HUDL/HUDL.cpp

```diff
@@ -19,43 +20,94 @@ namespace log = EVT::core::log;
 namespace HUDL {
 HUDL::HUDL(IO::GPIO& reg_select, IO::GPIO& reset, IO::SPI& spi) : lcd(DEV::LCD(reg_select, reset, spi)) {}
 
-void HUDL::dataWrite(uint8_t data) {
-    lcd.dataWrite(data);
+void HUDL::initLCD() {
+    lcd.initLCD();
+    lcd.clearLCD();
+
+    char* titles[9] = {
+        "B Voltage",
+        "Speed",
+        "RPM",
+        "Temp 1",
+        "Temp 2",
+        "Temp 3",
+        "Status 1",
+        "Pre Stat",
+        "Torque",
+    };
+
+    setDefaultSections(titles);
+    displaySectionHeaders();
+}
+
+CO_OBJ_T* HUDL::getObjectDictionary() {
+    return &objectDictionary[0];
+}
+
+uint16_t HUDL::getObjectDictionarySize() const {
+    return OBJECT_DICTIONARY_SIZE;
+}
+
+void HUDL::updateLCD() {
+    char tempOne[32];
+    std::sprintf(tempOne, "%lu C", *(this->thermTemps + 0));
+
+    char tempTwo[32];
+    std::sprintf(tempTwo, "%lu C", *(this->thermTemps + 1));
+
+    char tempThree[32];
+    std::sprintf(tempThree, "%lu C", *(this->thermTemps + 2));
+
+    char tempFour[32];
+    std::sprintf(tempFour, "%lu C", *(this->thermTemps + 3));
+
+    char voltage[32];
+    std::sprintf(voltage, "%lu v", totalVoltage);
+
+    setTextForSection(0, voltage);
+    //    setTextForSection(2, "25 MPH");
+    //    setTextForSection(2, "3000");
+    setTextForSection(3, tempOne);
+    setTextForSection(4, tempTwo);
+    setTextForSection(5, tempThree);
+    //    setTextForSection(6, "ON");
+    //    setTextForSection(7, "Ready");
+    //    setTextForSection(8, "100 NM");
 }
 
-void HUDL::commWrite(uint8_t data) {
-    lcd.commandWrite(data);
+void HUDL::setTextForSection(uint8_t section, const char* text) {
```

Comment:
Does the HUDL really need to expose all this functionality?  Can some of these functions be made private or deleted?